taca/illumina/HiSeqX_Runs.py

```diff
@@ -0,0 +1,379 @@
+                    fcd.write(_generate_clean_samplesheet(ssparser, fields_to_remove=['index2'], rename_samples=True, rename_qPCR_suffix = True, fields_qPCR=['SampleName']))
+            except Exception as e:
+                logger.error(e.text)
+                return False
+            logger.info(("Created SampleSheet.csv for Flowcell {} in {} ".format(self.id, samplesheet_dest)))
+        ##SampleSheet.csv generated
+        ##when demultiplexing SampleSheet.csv is the one I need to use
+        self.runParserObj.samplesheet  = SampleSheetParser(os.path.join(self.run_dir, "SampleSheet.csv"))
+
+        per_lane_base_masks = self._generate_per_lane_base_mask()
+        max_different_base_masks =  max([len(per_lane_base_masks[base_masks]) for base_masks in per_lane_base_masks])
+        if max_different_base_masks > 1:
+            # in a HiSeqX run I cannot have different index sizes in the SAME lane
+            logger.error("In FC {} found one or more lane with more than one base mask (i.e., different index sizes in \
+                         in the same lane".format(self.id))
+            return False
+        #I have everything to run demultiplexing now.
+        logger.info('Building bcl2fastq command')
+
+        with chdir(self.run_dir):
+            cl = [self.CONFIG.get('bcl2fastq')['bin']]
+            if self.CONFIG.get('bcl2fastq').has_key('options'):
+                cl_options = self.CONFIG['bcl2fastq']['options']
+                # Append all options that appear in the configuration file to the main command.
+                for option in cl_options:
+                    if isinstance(option, dict):
+                        opt, val = option.items()[0]
+                        cl.extend(['--{}'.format(opt), str(val)])
+                    else:
+                        cl.append('--{}'.format(option))
+            #now add the base_mask for each lane
+            for lane in sorted(per_lane_base_masks):
+                #iterate thorugh each lane and add the correct --use-bases-mask for that lane
+                #there is a single basemask for each lane, I checked it a couple of lines above
+                base_mask = [per_lane_base_masks[lane][bm]['base_mask'] for bm in per_lane_base_masks[lane]][0] # get the base_mask
+                base_mask_expr = "{}:".format(lane) + ",".join(base_mask)
+                cl.extend(["--use-bases-mask", base_mask_expr])
+
+            logger.info(("BCL to FASTQ conversion and demultiplexing started for "
+                 " run {} on {}".format(os.path.basename(self.id), datetime.now())))
+            misc.call_external_command_detached(cl, with_log_files=True)
+        return True
+
+
+    def check_run_status(self):
+        """
+           This function checks the status of a run while in progress. In the future will print the status
+        """
+        run_dir    =  self.run_dir
+        dex_status =  self.get_run_status()
+        return None
```

Comment:
I know, this method makes sense in HiSeq but not here. It made sense when we were computing the undetermined.
My idea is that flowcell parser will parse the file CycleTImes which tell you where you ar ein the sequencing, and this method will output the current state. This method can be used by Hugin also (we will talk about this @kate-v-stepanova ).
So @Galithil this is a work in progress, for now in the HiSeqX it does nothing important, but it will